FieldOpt/Reservoir/grid/eclgrid.cpp

```diff
@@ -66,58 +75,99 @@ namespace Reservoir {
-                dims.nz = eclDims.nz;
+                ERTWrapper::ECLGrid::ECLGridReader::Dims
+                        eclDims = ecl_grid_reader_->Dimensions();
+                        dims.nx = eclDims.nx;
+                        dims.ny = eclDims.ny;
+                        dims.nz = eclDims.nz;
                 return dims;
             }
-            else throw std::runtime_error("ECLGrid::Dimensions: Grid source must be defined before getting grid dimensions.");
+            else{
+                throw runtime_error("ECLGrid::Dimensions(): Grid "
+                                    "source must be defined before "
+                                    "getting grid dimensions.");
+            }
         }
 
         Cell ECLGrid::GetCell(int global_index)
         {
-            if (!IndexIsInsideGrid(global_index)) throw std::runtime_error("Error getting grid cell. Global index is outside grid.");
+            // Check if global index cell is inside overall (i.e., active+inactive) grid
+            if (!IndexIsInsideGrid(global_index)) {
+                throw runtime_error("ECLGrid::GetCell(int global_index): Error getting "
+                                    "grid cell. Global index is outside grid.");
+            }
+
             if (type_ == GridSourceType::ECLIPSE) {
-                ERTWrapper::ECLGrid::ECLGridReader::Cell ertCell = ecl_grid_reader_->GetGridCell(global_index);
+                ERTWrapper::ECLGrid::ECLGridReader::Cell
+                        ertCell = ecl_grid_reader_->GetGridCell(global_index);
 
-                // Get IJK index
-                ERTWrapper::ECLGrid::ECLGridReader::IJKIndex ecl_ijk_index = ecl_grid_reader_->ConvertGlobalIndexToIJK(global_index);
-                IJKCoordinate ijk_index = IJKCoordinate(ecl_ijk_index.i, ecl_ijk_index.j, ecl_ijk_index.k);
+                // Get IJK index corresponding to global index
+                ERTWrapper::ECLGrid::ECLGridReader::IJKIndex
+                        ecl_ijk_index = ecl_grid_reader_->ConvertGlobalIndexToIJK(global_index);
+                IJKCoordinate ijk_index = IJKCoordinate(ecl_ijk_index.i,
+                                                        ecl_ijk_index.j,
+                                                        ecl_ijk_index.k);
 
-                // Get center coordinates
+                // Get cell center coordinates
                 auto center = Eigen::Vector3d(ertCell.center);
 
-                // Get the corners
-                std::vector<Eigen::Vector3d> corners;
+                // Get cell corners
+                vector<Eigen::Vector3d> corners;
                 for (auto corner : ertCell.corners) {
                     corners.push_back(corner);
                 }
-                return Cell(global_index, ijk_index,
-                            ertCell.volume, ertCell.porosity, ertCell.permx, ertCell.permy, ertCell.permz,
+
+                // Return cell info
+                return Cell(global_index,
```

Comment:
In general, there is no need to put every argument on a new line. For instance, all the permeabilities could be on the same line. Longer code (in the vertical direction) is harder to read.